File: cadastro-moradores-api/src/main/java/com/elbertribeiro/cadastromoradores/telefone/Telefone.java
package com.elbertribeiro.cadastromoradores.telefone;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Telefone {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String numero;

    public Telefone() {
    }

    public Telefone(String numero) {
        this.numero = numero;
    }

    public Long getId() {
        return id;
    }

    public String getNumero() {
        return numero;
    }
}

File: cadastro-moradores-api/src/main/kotlin/com/elbertribeiro/cadastromoradores/telefone/Telefone.kt
package com.elbertribeiro.cadastromoradores.telefone

import javax.persistence.*

@Entity
class Telefone {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    val id: Long = 0
    val numero: String = ""
}

fun Telefone.toTelefoneDto() = TelefoneDto(
    numero = numero
)